src/compiler/checker.ts

```diff
@@ -23326,13 +23326,39 @@ namespace ts {
                             }
                             else {
                                 const middleLength = targetArity - startLength - endLength;
-                                if (middleLength === 2 && elementFlags[startLength] & elementFlags[startLength + 1] & ElementFlags.Variadic && isTupleType(source)) {
-                                    // Middle of target is [...T, ...U] and source is tuple type
-                                    const targetInfo = getInferenceInfoForType(elementTypes[startLength]);
-                                    if (targetInfo && targetInfo.impliedArity !== undefined) {
-                                        // Infer slices from source based on implied arity of T.
-                                        inferFromTypes(sliceTupleType(source, startLength, endLength + sourceArity - targetInfo.impliedArity), elementTypes[startLength]);
-                                        inferFromTypes(sliceTupleType(source, startLength + targetInfo.impliedArity, endLength), elementTypes[startLength + 1]);
+                                if (middleLength === 2) {
+                                    if (elementFlags[startLength] & elementFlags[startLength + 1] & ElementFlags.Variadic && isTupleType(source)) {
+                                        // Middle of target is [...T, ...U] and source is tuple type
+                                        const impliedArity = getInferenceInfoForType(elementTypes[startLength])?.impliedArity;
+                                        if (impliedArity !== undefined) {
+                                            // Infer slices from source based on implied arity of T.
+                                            inferFromTypes(sliceTupleType(source, startLength, endLength + sourceArity - impliedArity), elementTypes[startLength]);
+                                            inferFromTypes(sliceTupleType(source, startLength + impliedArity, endLength), elementTypes[startLength + 1]);
+                                        }
+                                    }

```

Comment:
this case was already here, it just got indented more since I put all 3 cases (1 old, 2 new ones) within the `if (middleLength === 2)` block